pkg/controllers/machinesync/machine_sync_controller_test.go

```diff
@@ -334,18 +388,90 @@ var _ = Describe("With a running MachineSync Reconciler", func() {
 				Expect(k8sClient.Create(ctx, capiMachine)).Should(Succeed())
 			})
 
-			It("should not make any changes to the CAPI machine", func() {
-				resourceVersion := capiMachine.GetResourceVersion()
-				Consistently(k.Object(capiMachine), timeout).Should(
-					HaveField("ResourceVersion", Equal(resourceVersion)),
-				)
+			Context("and there is no CAPI machineset owning the machine", func() {
+				It("should not make any changes to the CAPI machine", func() {
+					resourceVersion := capiMachine.GetResourceVersion()
+					Consistently(k.Object(capiMachine), timeout).Should(
+						HaveField("ResourceVersion", Equal(resourceVersion)),
+					)
+				})
+
+				It("should not create a MAPI machine", func() {
+					Consistently(k.ObjectList(&machinev1beta1.MachineList{}), timeout).ShouldNot(HaveField("Items",
+						ContainElement(HaveField("ObjectMeta.Name", Equal(capiMachine.GetName()))),
+					))
+				})
 			})
 
-			It("should not create a MAPI machine", func() {
-				Consistently(k.ObjectList(&machinev1beta1.MachineList{}), timeout).ShouldNot(HaveField("Items",
-					ContainElement(HaveField("ObjectMeta.Name", Equal(capiMachine.GetName()))),
-				))
+			Context("And there is a CAPI Machineset owning the machine", func() {
+				var capiMachineSetBuilder capiv1resourcebuilder.MachineSetBuilder
+				var capiMachineSet *capiv1beta1.MachineSet
+				BeforeEach(func() {
+					By("Creating the CAPI machine set")
+					capiMachineSet = capiMachineSetBuilder.
+						WithNamespace(capiNamespace.GetName()).
+						WithName("foo-ms").Build()
+
+					Expect(k8sClient.Create(ctx, capiMachineSet)).Should(Succeed())
+
+					By("Setting owner references on the CAPI machine")
+					capiMachine.SetOwnerReferences([]metav1.OwnerReference{
+						{
+							APIVersion: capiv1beta1.GroupVersion.String(),
+							Kind:       machineSetKind,
+							Name:       capiMachineSet.GetName(),
+							UID:        "1349b459-482e-44bd-8b22-dc53a0d98d7c",
+						},
+					})
+
+					Expect(k8sClient.Update(ctx, capiMachine)).Should(Succeed())
+				})
+
+				Context("with no MAPI counterpart", func() {
+					It("should not make any changes to the CAPI machine", func() {
+						resourceVersion := capiMachine.GetResourceVersion()
+						Consistently(k.Object(capiMachine), timeout).Should(
+							HaveField("ResourceVersion", Equal(resourceVersion)),
+						)
+					})
+
+					It("should not create a MAPI machine", func() {
+						Consistently(k.ObjectList(&machinev1beta1.MachineList{}), timeout).ShouldNot(HaveField("Items",
+							ContainElement(HaveField("ObjectMeta.Name", Equal(capiMachine.GetName()))),
+						))
+					})
+				})
+
```

Comment:
Once we have owner reference conversion in place, this test case won't fail and can be un-commented / built out.